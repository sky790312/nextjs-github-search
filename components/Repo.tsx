/* eslint-disable @typescript-eslint/camelcase */
import styled from 'styled-components'
import { GithubRepoSchema } from '@/api/schema'
import Link from 'next/link'
import { useRouter } from 'next/router'

export interface Props {
  repo: GithubRepoSchema
}

export const Repo: React.FC<Props> = ({ repo }) => {
  const router = useRouter()
  const { userName } = router.query

  return (
    <Link href={`/repo?userName=${userName}&repoName=${repo.name}`}>
      <RepoWrapper>
        <div>
          <h3>{repo.name}</h3>
          <p>Star: {repo.stargazers_count}</p>
        </div>
      </RepoWrapper>
    </Link>
  )
}

const RepoWrapper = styled.div`
  margin: 10px;
  padding: 20px;
  cursor: pointer;
  background-color: ${props => props.theme.colors.gray};

  &:hover {
    opacity: 0.7;
  }

  @media (min-width: 1140px) {
    display: flex;
    align-items: center;
  }
`
