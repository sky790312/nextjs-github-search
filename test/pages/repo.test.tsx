import React from 'react'
import { render, withProviders } from '@/test/testUtils'
import Repo from '@/pages/repo'

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '',
      pathname: '',
      query: '',
      asPath: '',
    }
  },
}))

describe('Repo page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(withProviders(<Repo />), {})
    expect(asFragment()).toMatchSnapshot()
  })
})
