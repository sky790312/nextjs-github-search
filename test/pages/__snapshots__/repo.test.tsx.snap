// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="GlobalStyles__Container-sc-1m5drgn-0 frTxMm"
  >
    <h1 />
    <span
      class="Spinner__StyledSpinner-sc-12ylinb-0 berLTj"
    />
    <div
      class="GlobalStyles__ListContainer-sc-1m5drgn-2 lDTWr"
    />
    <hr />
    <div />
  </div>
</DocumentFragment>
`;
